cmd/dockerd/daemon.go

```diff
@@ -244,6 +245,13 @@ func (cli *daemonCLI) start(ctx context.Context) (err error) {
 
 	tp := newTracerProvider(ctx)
 	otel.SetTracerProvider(tp)
+
+	// Workaround OTEL memory leak
+	// See: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/5190
+	// The need for this workaround is checked by the TestOtelMeterLeak test
+	// TODO: Remove this workaround after upgrading to v1.30.0
+	otel.SetMeterProvider(noop.MeterProvider{})
+
```

Comment:
Curious if this should be "as early as possible" in the chain (as a default, which can be overridden by "whatever" will override).

Would this fix still work if we'd put it (e.g.) in `newDaemonCommand` (assuming setting it in `main()` wouldn't make it easy to test)

Reply:
It would work, but I think it's much better to have it in the same place as other OTEL related stuff. If we'd like to actually override it, we can just remove it as it won't be needed anyway.

Reply:
Yeah, it's also that we don't always know where things are overridden (at least I recall some of the ugly bits where BuildKit setup things in an `init()`.

> but I think it's much better to have it in the same place as other OTEL related stuff


Yeah, even wondering if there's bits around that that we should also set up earlier (making sure that we trace as much as possible, starting with `main` ?)

Reply:
I can move it to main, so at least it stops the leak earlier. Although that shouldn't be the case as AFAIK the only meter counting on our side come from `otelhttp`.

I don't think it's reasonable for us to consider dependencies silently override the meter provider without our deliberate decision though.

Reply:
Agreed.